buggy_oop.py

```
@@ -0,0 +1,83 @@
+import re
+from datetime import datetime
+
+class User:
+    def __init__(self, name, email, age):
+        # Bug fixed: add input validation in constructor
+        if not name or not name.strip():
+            raise ValueError("Name cannot be empty")
+        if not self._is_valid_email(email):
+            raise ValueError("Invalid email format")
+        if not isinstance(age, int) or age < 0:
+            raise ValueError("Age must be a non-negative integer")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)